src/midi/midi.cpp

```diff
@@ -617,135 +668,165 @@ class MIDI final {
 		}
 
 		trim(midiconfig_prefs);
-		const char* midiconfig = midiconfig_prefs.c_str();
-
-		auto open_handler = [&](MidiHandler* handler) -> bool {
-			const auto opened = handler && handler->Open(midiconfig);
-			if (opened) {
-				midi.is_available = true;
-				midi.handler      = handler;
-				LOG_MSG("MIDI: Opened device: %s",
-				        handler->GetName().c_str());
-			}
-			return opened;
-		};
+		const auto midiconfig_pref = midiconfig_prefs.c_str();
```

Comment:
This copying of the std::string prefs to a new `char*` is probably  unnecessary because the `try_create_auto_device(...)` and `create_device(...)` both take in std::string&, so should be good to go passing in the trimmed `midiconfig_prefs`.

Reply:
Indeed! Probably the remnants of some refactoring steps. Simplified! 🛠️ 😄